bin/manifest.py

```
@@ -0,0 +1,244 @@
+    seed = os.urandom(16)
+    return seed
+
+def gen_ed():
+    sk = os.urandom(32)
+    pk = ed25519.publickey(sk)
+    return sk, pk
+
+def gen_ec():
+    sk = ecdsa.SigningKey.generate(curve=ecdsa.SECP256k1)
+    vk = sk.get_verifying_key()
+    sig = sk.sign("message")
+    assert vk.verify(sig, "message")
+    return sk, vk
+
+def gen_manifest(pk, vpk, seq):
+    s = ""
+    s += sfSequence
+    s += str32(seq)
+    s += sfPublicKey
+    s += strvl(pk)
+    s += sfSigningPubKey
+    s += strvl(vpk)
+    return s
+
+def sign_manifest(m, sk, pk):
+    s = "MAN\0"
+    s += m
+    sig = ed25519.signature(s, sk, pk)
+    m += sfSignature + strvl(sig)
+    return m
+
+#-----------------------------------------------------------
+
+def wrap(s, cols = 60):
+    n = len(s)
+    if (n <= cols):
+        return s
+    l = (n + cols - 1) / cols
+    w = n / l
+    while(l > 0):
+        s = s[:w * l] + '\n' + s[w*l:]
+        l -= 1
+    return s
+
+#-----------------------------------------------------------
+
+if (len(sys.argv) == 2 and sys.argv[1] == 'create'):
+    sk, pk = gen_ed()
+    apk = chr(0xed) + pk
+    pkh = ra_encode(VER_NODE_PUBLIC, apk)
+    skh = ra_encode(VER_NODE_PRIVATE, sk)
+    v0, apk0 = ra_decode(pkh)
+    assert v0 == VER_NODE_PUBLIC
+    assert apk0 == apk
+    print ("[validators]")
+    print (pkh)
+    print
+    print ("[master_secret]")
+    print (skh)
+    exit()
+
+if (len(sys.argv) == 3 and sys.argv[1] == 'check'):
+    ver, b = ra_decode(sys.argv[2])
+    print ('ver = ' + str(ver))
+    print ('len = ' + str(len(b)))
+    h = ra_encode(ver, b)
+    print (h)
+    assert h == sys.argv[2]
+    exit()
+
+if (len(sys.argv) == 5 and sys.argv[1] == 'sign'):
+    seq = int(sys.argv[2])
+    vpkh = sys.argv[3]
+    skh = sys.argv[4]
+    try:
+        v, avpk = ra_decode(vpkh)
+        if (v != VER_NODE_PUBLIC or
+            len(avpk) != 33 or
+            (ord(avpk[0]) != 2 and ord(avpk[0]) != 3)):
+            raise ValueError()
```

Why not actually raise the error, and not catch it?  it's a failure, yes?